### PR TITLE
Filter transcripts by Ensembl ID, Ensembl gene ID, HGNC gene ID, HGNC gene symbol

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Filter database transcripts by Ensembl ID, Ensembl gene ID, gene HGNC ID, gene HGNC symbol

### How to test:
- Launch a demo instance and filter transcripts by the items above

### Expected outcome:
- [x] Results should be returned by the queries

### Review:
- [ ] Code approved by
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -97,13 +109,13 @@ def _override_get_db():
     return TestClient(app)
 
 
-@pytest.fixture(name="demo_session", scope="module")
```
Comment (northwestwitch) [inline tests/conftest.py:100]: I had to revert this, as tests stopped working

**Comment by northwestwitch** [comment]
Transcripts by Ensembl ID 🆗 

image

Transcripts by Ensembl GENE ID: 🆗 

image

Transcripts by HGNC gene ID: 🆗 

image


Transcripts by HGNC gene symbol: 🆗 

image

**Comment by northwestwitch** [comment]
This is also done!

File: src/chanjo2/crud/intervals.py [hunk 2]
```diff
@@ -168,16 +177,50 @@ def bulk_insert_transcripts(db: Session, transcripts: List[TranscriptBase]):
...
+    ensembl_ids: Optional[List[str]],
+    hgnc_ids: Optional[List[int]],
+    hgnc_symbols: Optional[List[str]],
+    ensembl_gene_ids: Optional[List[str]],
+    limit: Optional[int],
+) -> List[SQLTranscript]:
+    """Return transcripts according to specified fields."""
+
+    transcripts: query.Query = db.query(SQLTranscript)
+    genes = db.query(SQLGene)
```
Comment (henrikstranneheim) [inline src/chanjo2/crud/intervals.py:192]:
```suggestion
    genes: query.Query = db.query(SQLGene)
```

File: src/chanjo2/crud/intervals.py [hunk 3]
```diff
@@ -56,7 +65,7 @@ def _filter_intervals_by_hgnc_symbols(
     interval_type: Union[SQLGene, SQLTranscript, SQLExon],
     hgnc_symbols: List[str],
 ) -> List[Union[SQLGene, SQLTranscript, SQLExon]]:
```
Comment (henrikstranneheim) [inline src/chanjo2/crud/intervals.py:67]:
```suggestion
) -> query.Query:
```
I do not think you are reading the db yet and interpolating to real objects

File: src/chanjo2/crud/intervals.py [hunk 4]
```diff
@@ -42,12 +42,21 @@ def _filter_intervals_by_ensembl_ids(
...
+) -> List[Union[SQLGene, SQLTranscript, SQLExon]]:
+    """Filter intervals using a list of Ensembl gene IDs."""
+    return intervals.filter(interval_type.ensembl_gene_id.in_(ensembl_gene_ids))
+
+
 def _filter_intervals_by_hgnc_ids(
     intervals: query.Query,
     interval_type: Union[SQLGene, SQLTranscript, SQLExon],
     hgnc_ids: List[int],
 ) -> List[Union[SQLGene, SQLTranscript, SQLExon]]:
```
Comment (henrikstranneheim) [inline src/chanjo2/crud/intervals.py:58]:
```suggestion
) -> query.Query:
```
I do not think you are reading the db yet and interpolating to real objects

File: src/chanjo2/crud/intervals.py [hunk 5]
```diff
@@ -42,12 +42,21 @@ def _filter_intervals_by_ensembl_ids(
     return intervals.filter(interval_type.ensembl_id.in_(ensembl_ids))
 
 
+def _filter_intervals_by_ensembl_gene_ids(
+    intervals: query.Query,
+    interval_type: Union[SQLGene, SQLTranscript, SQLExon],
+    ensembl_gene_ids: List[str],
+) -> List[Union[SQLGene, SQLTranscript, SQLExon]]:
```
Comment (henrikstranneheim) [inline src/chanjo2/crud/intervals.py:49]:
```suggestion
) -> query.Query:
```
I do not think you are reading the db yet and interpolating to real objects

File: src/chanjo2/crud/intervals.py [hunk 6]
```diff
@@ -168,16 +177,50 @@ def bulk_insert_transcripts(db: Session, transcripts: List[TranscriptBase]):
...
+    if hgnc_ids:
+        genes: query.Query = _filter_intervals_by_hgnc_ids(
+            intervals=genes, interval_type=SQLGene, hgnc_ids=hgnc_ids
+        ).all()
+    elif hgnc_symbols:
+        genes: query.Query = _filter_intervals_by_hgnc_symbols(
+            intervals=genes, interval_type=SQLGene, hgnc_symbols=hgnc_symbols
+        ).all()
+    if hgnc_ids or hgnc_symbols:
+        ensembl_gene_ids = [gene.ensembl_id for gene in genes]
```
Comment (henrikstranneheim) [inline src/chanjo2/crud/intervals.py:202]:
```suggestion
        ensembl_gene_ids: List[str] = [gene.ensembl_id for gene in genes]
```

**Review by henrikstranneheim — APPROVED** [review]
Great work 👍

File: src/chanjo2/crud/intervals.py (hunk 4 above)
Comment (northwestwitch) [reply src/chanjo2/crud/intervals.py:58]: Yes, you're right!

**Pull request merged**